### DISCUSSION TITLE
Deactivate SQL checks in Codacy

### DISCUSSION THREAD

**Discussion by solth** [kitodo/kitodo-production#4418] [General]
At the moment, Codacy checks the syntax of SQL files using SQLint. SQLint only supports "ANSI SQL" syntax:

> At this stage, SQLint checks SQL against the ANSI syntax, and uses the PostgreSQL SQL parser to achieve this. SQLint does not have support for non-standard SQL variants (e.g. MySQL), but contributions are welcome.

(see https://github.com/purcell/sqlint , "About"). 

This leads to SQL statements that are perfectly fine in our environment to be marked as syntax errors by Codacy:
<img width="519" alt="Bildschirmfoto 2021-05-19 um 09 58 48" src="https://user-images.githubusercontent.com/19183925/118776964-db586b00-b888-11eb-9ada-128a0c84e126.png">
Since this imposes unnecessary constraints on our development I would propose to deactive the SQL checks in Codacy until SQLint supports checking the syntax of the SQL extensions we use.
@Kathrin-Huber , @henning-gerhardt , @matthias-ronge : what do you think?

**Comment by matthias-ronge** [comment]
If that's not just a configuration problem (i.e., we would have to explicitly state that, for example, the syntax accepted by MariaDB is considered valid) then I'm for it. I have never understood these "mistakes".
